MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClient.java

```diff
@@ -24,1103 +24,21 @@
-import com.shopify.buy.model.Collection;
-import com.shopify.buy.model.Collection.SortOrder;
-import com.shopify.buy.model.CreditCard;
-import com.shopify.buy.model.Customer;
-import com.shopify.buy.model.CustomerToken;
-import com.shopify.buy.model.GiftCard;
-import com.shopify.buy.model.Order;
-import com.shopify.buy.model.Payment;
-import com.shopify.buy.model.PaymentSession;
-import com.shopify.buy.model.internal.PaymentRequest;
-import com.shopify.buy.model.Product;
-import com.shopify.buy.model.ShippingRate;
-import com.shopify.buy.model.Shop;
-import com.shopify.buy.model.internal.AccountCredentialsWrapper;
-import com.shopify.buy.model.internal.AddressWrapper;
-import com.shopify.buy.model.internal.AddressesWrapper;
-import com.shopify.buy.model.internal.CheckoutWrapper;
-import com.shopify.buy.model.internal.CollectionListings;
-import com.shopify.buy.model.internal.CreditCardWrapper;
-import com.shopify.buy.model.internal.CustomerTokenWrapper;
-import com.shopify.buy.model.internal.CustomerWrapper;
-import com.shopify.buy.model.internal.EmailWrapper;
-import com.shopify.buy.model.internal.GiftCardWrapper;
-import com.shopify.buy.model.internal.MarketingAttribution;
-import com.shopify.buy.model.internal.OrderWrapper;
-import com.shopify.buy.model.internal.OrdersWrapper;
-import com.shopify.buy.model.internal.PaymentRequestWrapper;
-import com.shopify.buy.model.internal.PaymentWrapper;
-import com.shopify.buy.model.internal.ProductListings;
-import com.shopify.buy.model.internal.ShippingRatesWrapper;
-import com.shopify.buy.utils.CollectionUtils;
-
-import java.io.IOException;
-import java.nio.charset.Charset;
-import java.util.List;
-import java.util.concurrent.TimeUnit;
-
-import okhttp3.Interceptor;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import retrofit2.Response;
-import retrofit2.Retrofit;
-import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
-import retrofit2.converter.gson.GsonConverterFactory;
-import rx.Observable;
-import rx.Scheduler;
-import rx.android.schedulers.AndroidSchedulers;
-import rx.functions.Action1;
-import rx.functions.Func1;
-import rx.schedulers.Schedulers;
 
 /**
- * The {@code BuyClient} provides all requests needed to perform request on the Shopify Checkout API. Use this class to perform tasks such as getting a shop, getting collections and products for a shop, creating a {@link Checkout} on Shopify and completing Checkouts.
- * All API methods presented here run asynchronously and return results via callback on the Main thread.
+ * Client responsible for all requests needed to perform request on the Shopify Checkout API.
+ * Use this class to perform tasks such as getting a shop, getting collections and products for a shop,
+ * creating a {@link Checkout} on Shopify and completing Checkouts.
+ * All API methods presented here run asynchronously and return results via callback or Rx observables on the callback scheduler thread.
  */
-public class BuyClient implements IBuyClient {
```

Comment:
I can see why we need the IBuyClient naming on the interface because we do not want to switch BuyClient from a Class to an interface, but I am not a huge fan of prefixing "I".

Maybe BuyClientInterface? or BuyProvider?  not sure what makes sense here though.